.circleci/config.yml

```diff
@@ -194,21 +194,21 @@ workflows:
           name: ubuntu-focal
           context: openquantumsafe
           IMAGE: openquantumsafe/ci-ubuntu-focal-x86_64:latest
-          CMAKE_ARGS: -DOQS_STRICT_WARNINGS=ON -DOQS_USE_OPENSSL=OFF
+          CMAKE_ARGS: -DOQS_STRICT_WARNINGS=ON -DOQS_USE_OPENSSL=OFF -DBUILD_SHARED_LIBS=ON
           OPENSSL_PREINSTALL: openssl@1
       - ubuntu:
           name: ubuntu-jammy
           context: openquantumsafe
           IMAGE: openquantumsafe/ci-ubuntu-jammy:latest
-          CMAKE_ARGS: -DOQS_STRICT_WARNINGS=ON -DOQS_ALGS_ENABLED=STD
+          CMAKE_ARGS: -DOQS_STRICT_WARNINGS=ON -DOQS_ALGS_ENABLED=STD -DBUILD_SHARED_LIBS=ON
```

Comment:
Why adding the shared option? I think this test was meant to build/test a static version (judging from the other test's name "...-shared" :-)

Reply:
That's a good point. https://github.com/open-quantum-safe/oqs-provider/commit/8a0bca25f0aeabdf3a2b9f2bfce5c236c7b377c6 changed the build to "shared" without also adapting the (meant-to-be-static) CI tests.
That said, that commit seems to be completely improper, reading [cmake documentation on 
add_library](https://cmake.org/cmake/help/latest/command/add_library.html): The library type should be "MODULE" and not "SHARED" as the documentation reads

> MODULE libraries are plugins that are not linked into other targets but may be loaded dynamically at runtime using dlopen-like functionality. 

Thus, I'd suggest reverting this back to "MODULE": https://github.com/open-quantum-safe/oqs-provider/blob/6b3483965293db2a3a1dc607468f82739247958c/oqsprov/CMakeLists.txt#L34

Objections, @bhess? Would you want to do this in this PR, @thb-sb ?